### PR TITLE
update SPDX license headers in source code

### PR DESCRIPTION
**Pull request by merobi-hub**
Signed-off-by: Michael Robinson <merobi@gmail.com>

### Problem

The SPDX license headers, which we added last year, are out of date now that it's 2023. So they need updating to be current.

### Solution

This will change 2022 in the headers to 2023 to keep them current.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/proposal.md [hunk 1]
```diff
@@ -14,3 +14,7 @@ Please describe the corresponding use cases.
 **Proposed implementation**
 This section describes *how* you propose to model it. 
 If you are you proposing a new facet, please mention its name and schema.
+
+----
+SPDX-License-Identifier: Apache-2.0\
+Copyright 2018-2023 contributors to the OpenLineage project
```
Comment (mobuchowski) [inline .github/ISSUE_TEMPLATE/proposal.md:20]: I don't think this should be in template?

File: client/python/README.md [hunk 2]
```diff
@@ -70,4 +70,4 @@ This way of configuring the client supports only `http` transport, and only a su
 
 ----
 SPDX-License-Identifier: Apache-2.0\
-Copyright 2018-2022 contributors to the OpenLineage project
\ No newline at end of file
+Copyright 2018-2023 contributors to the OpenLineage project
```
Comment (mobuchowski) [inline client/python/README.md:73]: I don't think we need this license identifier in .md file either

File: integration/spark/databricks/README.md [hunk 3]
```diff
@@ -52,3 +50,7 @@ You'll see a message in the Log4j output that looks like the following snippet.
 ```
 YY/MM/DD HH:mm:ss INFO OpenLineageContext: Lineage completed successfully: ResponseMessage(responseCode=200, body=null, error=null) <... LINEAGE INFO ...>
 ```
+
+----
+SPDX-License-Identifier: Apache-2.0\
+Copyright 2018-2023 contributors to the OpenLineage project
```
Comment (mobuchowski) [inline integration/spark/databricks/README.md:56]: I don't think we need this here

**Review by mobuchowski — CHANGES_REQUESTED** [review]
I think this needs few changes around templates and .md files, but overall giant majority of files is fine @merobi-hub

**Comment by merobi-hub** [comment]
Thanks for the feedback @mobuchowski . It's ready for another look now.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**